### ISSUE TITLE
trying to push subrepo changes to branch: Can't find a common ancestor ...

### ISSUE THREAD

**Issue by brianjmurrell** [ingydotnet/git-subrepo#345]
I'm trying to push the changes I have made to a subrepo in a project back up to the repo that the subrepo was created from and getting that annoying "Can't find a common ancestor" error:

```
$ git subrepo push include -b travis-tools
git-subrepo: Can't find a common ancestor between refs/subrepo/include/fetch and subrepo-push/include. This might be the case if someone used push --force on the subrepo. You need to find out why and possibly reclone the subrepo.
```

Nobody has done any `push --force` on the repo that is the subrepo.

I have gotten this error before trying to pull changes from the subrepo also but ended up just giving up and recloning the subrepo.  But obviously, if I'm going to have to do that every time I want to do something, I may as well just stop using subrepo.

Some information you will probably want:

```
$ cat include/.gitrepo 
; DO NOT EDIT (unless you know what you are doing)
;
; This subdirectory is a git "subrepo", and this file is maintained by the
; git-subrepo command. See https://github.com/git-commands/git-subrepo#readme
;
[subrepo]
	remote = git@github.com:intel-hpdd/module-tools.git
	branch = master
	commit = ae6b9b5c2ef099690d0b0f3bd8ee184023f5c8a5
	parent = a9dec59277c3caa239641e615828b885753d82fe
	cmdver = 0.3.1
```

The goal here is to push the changes to my subrepo up to a branch in _github.com:intel-hpdd/module-tools.git_ so that a PR can be made from it which can be reviewed and landed to master.

**Comment by karptonite** [comment]
I can confirm that I am also having this problem. Bizarrely, I've "solved" it by cloning my main repo, and doing the `git subrepo push` from the clone. While the two repos are ostensibly identical, they do not behave the same with regard to this issue.

**Comment by brianjmurrell** [comment]
So, I ended up adding all of the commits from my subrepo to a clone of the subrepo's repo and pushing from there (also  -- if that's what you are describing @karptonite -- I think it is).  Basically, sidestepping everything that subrepo is supposed to give me and dealing with the subrepo as a manual sub-repo.  :-(

**Comment by grimmySwe** [comment]
@brianjmurrell Could you test with the release/0.4.0 branch? The algorithm for 0.3.X version is a little bit flawed due to some assumptions made by me.

**Comment by vjpr** [comment]
I was having a similar issue and `release/0.4.0` solved it.

---

My use case for reference:

I had a sub dir that I wanted to extract to its own public repo.

I could not use `git subrepo init` because I wanted to squash the history to a single commit when extracting to avoid pushing private code in earlier commits to github. (Lesson learned was to always keep code that should not be public out of a dir you eventually want to make public.)

I manually copied the files from my sub dir to a new repo and pushed, then used `git subrepo clone <repo>`.

`git subrepo push foo` caused error `trying to push subrepo changes to branch: Can't find a common ancestor`. Fixed in `release/0.4.0`.
